src/ApcuRateLimiter.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+declare(strict_types=1);
+
+namespace RateLimit;
+
+final class ApcuRateLimiter extends AbstractElapsedTimeRateLimiter
```

Comment:
Sorry for chiming in, but may I ask why the class is defined as final?

Reply:
Sure, no prob

Final classes here makes a lot of sense as the functionality the limiters provide is really focused (the interfaces have only one method) and the reason for them to change is absolutely limited, or none at all. You can always decorate the classes

Besides [you should make your classes final](https://matthiasnoback.nl/2018/09/final-classes-by-default-why/) (good article)